### ISSUE TITLE
Error while running powershell command using cmd.powershell module

### ISSUE THREAD

**Issue by varun292006** [saltstack/salt#43902]
### Description of Issue/Question
SaltStack throws error when powershell command is executed  using cmd.powershell module
 
### Setup
1 ubuntu master
3 windows-7 minions

### Steps to Reproduce Issue
Create a state file as below:
```
{%- set var1 = salt.cmd.powershell("Get-WmiObject win32_service | where-object {$_.PathName -like \"C:\salt\*\"} | select-object -expandproperty 'Name'") %}
Print service name:
  module.run:
    - name: test.echo
    - text: Service name is --- '{{ var1 }}'

```

Execute the state file using the below command:
`salt -v -t 60  'finance' state.apply samples.run_cmd`

State file gets executed without any error but does not return the name of the service
```
Executing job with jid 20171004114008515705
-------------------------------------------

finance:
----------
          ID: Print service name
    Function: module.run
        Name: test.echo
      Result: True
     Comment: Module function test.echo executed
     Started: 11:40:18.977000
    Duration: 16.0 ms
     Changes:   
              ----------
              ret:
                  Service name is --- '{}'

Summary for finance
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  16.000 ms
```

Below trace can be seen on minion:
```
2017-10-04 11:40:15,966 [salt.fileclient  ][INFO    ][3872] Fetching file from saltenv 'base', ** done ** 'samples/run_cmd.sls'
2017-10-04 11:40:15,966 [salt.loaded.int.module.cmdmod][INFO    ][3872] Executing command 'Powershell -NonInteractive -NoProfile "Get-WmiObject win32_service | where-object {$_.PathName -like \"C:\salt\*\"} | select-object -expandproperty 'Name' | ConvertTo-JSON"' in directory 'C:\Windows\system32\config\systemprofile'
2017-10-04 11:40:18,976 [salt.loaded.int.module.cmdmod][ERROR   ][3872] Command 'Get-WmiObject win32_service | where-object {$_.PathName -like "C:\salt\*"} | select-object -expandproperty 'Name' | ConvertTo-JSON' failed with return code: 1
2017-10-04 11:40:18,976 [salt.loaded.int.module.cmdmod][ERROR   ][3872] output: The term 'ConvertTo-JSON' is not recognized as the name of a cmdlet, function, 
script file, or operable program. Check the spelling of the name, or if a path 
was included, verify that the path is correct and try again.
At line:1 char:131
+ Get-WmiObject win32_service | where-object {$_.PathName -like "C:\salt\*"} | 
select-object -expandproperty 'Name' | ConvertTo-JSON <<<< 
    + CategoryInfo          : ObjectNotFound: (ConvertTo-JSON:String) [], Comm 
   andNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
2017-10-04 11:40:18,976 [salt.loaded.int.module.cmdmod][ERROR   ][3872] Error converting PowerShell JSON return
Traceback (most recent call last):
  File "c:\salt\bin\lib\site-packages\salt\modules\cmdmod.py", line 3098, in powershell
    return json.loads(response)
  File "C:\Python27\Lib\json\__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "C:\Python27\Lib\json\decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python27\Lib\json\decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
2017-10-04 11:40:18,976 [salt.state       ][INFO    ][3872] Running state [test.echo] at time 11:40:18.977000
```

The command gives the result when executed manually on the powershell.

### Versions Report
Master:
```
Salt Version:
           Salt: 2017.7.1
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: 3.5.0
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.5
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.10.0-35-generic
         system: Linux
        version: Ubuntu 16.04 xenial

```

Minion:
```
Salt Version:
           Salt: 2017.7.1

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: 2.0.2
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.5
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.150
0 64 bit (AMD64)]
   python-gnupg: 0.4.0
         PyYAML: 3.11
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist:
         locale: cp1252
        machine: AMD64
        release: 7
         system: Windows
        version: 7 6.1.7601 SP1
```

**Comment by garethgreenaway** [comment]
@varun292006 Thanks for the report.  This seems to be an issue related to the version of Powershell that is installed.  The command ConvertTo-JSON is only available in PowerShell v3.  Can you verify what version you have installed?

**Comment by varun292006** [comment]
Please find the requested information below:

```
salt -t 60 'finance' cmd.shell_info powershell
jid: 20171005085514675987
finance:
    ----------
    applicationbase:
        C:\Windows\System32\WindowsPowerShell\v1.0
    consolehostassemblyname:
        Microsoft.PowerShell.ConsoleHost, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, ProcessorArchitecture=msil
    consolehostmodulename:
        C:\Windows\System32\WindowsPowerShell\v1.0\Microsoft.PowerShell.ConsoleHost.dll
    crlversion:
        2.0.50727
    installed:
        True
    path:
        C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe
    pscompatibleversions:
        - 1.0
        - 2.0
    psversion:
        2.0
    version:
        2.0
    version_raw:
        2.0
```

or,
```
PS C:\> $PSVersionTable.PSVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
2      0      -1     -1
```

**Comment by damon-atkins** [comment]
https://www.microsoft.com/en-au/download/details.aspx?id=34595  PowerShell 3 needs .NET4
https://www.microsoft.com/en-us/download/details.aspx?id=54616 PowerShell 5 needs .NET4.5

**Comment by varun292006** [comment]
@damon-atkins  Could you also please tell me what should I do with the links that you mentioned above?

Do you want me to upgrade the powershell version?

**Comment by garethgreenaway** [comment]
@varun292006 Thanks for the reporting information, the `ConvertTo-JSON` powershell command is not available in PowerShell 2.0, which is what you have installed.  It was added in PowerShell 3.0.  Are you able to upgrade to a newer version of PowerShell?  The links @damon-atkins provided indicate the version of the .NET packages that are required for those versions of PowerShell.

**Comment by maximd** [comment]
Would it be possible to display a warning when the minion detects that PowerShell 2.0 is installed (and that it should be upgraded at least to 3.0)? Version 2.0 is the default version of PowerShell installed on windows 7 systems and it can take quite a while to figure out that PowerShell should in fact be upgraded to let `cmd.powershell` actually work.

Also, stop me if I am wrong but it seems that the documentation at https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.cmdmod.html#salt.modules.cmdmod.powershell does not mention that PowerShell 3.0 should be installed to have `ConvertTo-JSON` :(
